torch/_dynamo/eval_frame.py

```diff
@@ -326,15 +326,26 @@ def __init__(self, backend=None):
 
     @property
     def example_inputs(self):
-        return clone_inputs(self.original_example_inputs)
+        if config.fake_tensor_propagation:
+            return clone_inputs(self.original_example_inputs_fake)
+        return clone_inputs(self.original_example_inputs_real)
 
-    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+    def __call__(self, gm: torch.fx.GraphModule, fake_inputs, real_inputs):
 
         self.restore = checkpoint_params(gm)
-        self.original_example_inputs = clone_inputs(example_inputs)
+        self.original_example_inputs_real = clone_inputs(real_inputs)
```

Comment:
```suggestion
        assert config.verify_correctness
        self.original_example_inputs_real = clone_inputs(real_inputs)
```
This clone will be expensive, so want to make sure we only hit it with this config flag set.

Reply:
Good call!